packages/framework/immutable-merge/README.md

```diff
@@ -3,82 +3,183 @@
 This package provides a relatively concise routine to handle merging multiple objects together with the following characteristics:
 
 - No modifications will be made to any object
-- The resulting object will have the minimum number of updates. If only one value is updated three levels deep, only that value and the chain of containing objects will be recreated.
+- Minimal updates. If only one value is updated three levels deep, only that value and the chain of containing objects will be recreated.
 - Empty objects or undefined objects will be ignored and not cause a new branch to be created.
 - Recursion is controllable in a variety of ways
 
-## IMergeOptions
+Note that this does not provide a **strict** immutable package on its own. It also doesn't operate or return only readonly objects. This is to provide flexibility. It could easily be wrapped in various ways to provide that type of functionality, but it is provided in a more flexible form to be useful in other scenarios as well.
```

Comment:
When you say it doesn't strictly operate on readonly/immutable objects, is that a consequence of the depth of the merge (some property of some property of some property might get mutated across the result and one of the initial objects after the merge)?

Reply:
I primarily mean that it doesn't have any enforcement.  The types are not required to be readonly and there aren't ways of hiding properties under functions like some libraries out there.  It just follows the principles above.